policy/policy.sample.json

```diff
@@ -0,0 +1,70 @@
+{
+    "SecurityPolicies": [
+        {
+            "Name": "policy1", # names can be same for multiple tenants thus use uuid.
+            "PolicyID": "3710940D-FBC0-4A44-A95D-E549E628B4DF", # unique identifier across tenants.
+            "Associations": [ # can attach multiple tenants associations in case you want blanket policy support.
+                {
+                    "Tenant":"tenant1", #optional if TenantID below is provided 
+                    "TenantID": "3A6E467F-370B-43EA-BCEC-C28912EDD752", # optional but good to have to relate things as names are not unique
+                    "Segment": "Segment1", # optional
+                    "SegmentID": "01CA06A0-429B-4EE1-9185-F17D5CE4D0F2",
+                },
+                {
+                    "Tenant":"tenant1",
+                    "Segment": "Segment2",
+                },
+            ],
+            "Tags": [
+                {
+                    "Key": "Platform",
+                    "Value": "Openstack",
+                },
+                {
+                    "Key": "TenantID",
+                    "Value": "D46FD18B-A381-4297-A6D4-9DA594EE71C8",
+                },
+            ], # meta data attached to policies for various external enviorments like openstack/kybernetes
+            "Direction" : "Egress", # can be Egress or Ingress.
+            "CidrBlock": "0.0.0.0/0", # IP from L3 header
```

Comment:
The "peer" in the conversation can be:

* CIDR block
* Segment (of same tenant)
* Segment (of other tenant)
* Entire same tenant
* Entire different tenant
* Entire different cluster
* Segment/tenant of different cluster
* etc.

Since CIDR block is not the only thing, I don't want that to appear at the top level.

Otherwise, you could randomly scatter "peer" specifiers across your top level of the policy definition and it would be a big mess to look at.

We could use "peer" as the top level element under which you then specify one (more than one?) of those peer specifiers. Problem is I don't particularly like the word "peer" for this, since it sounds too equal in a way. Is there a better word?

I don't share the fear of confusion about the meaning of 'src' and 'dst'. Since we have specific 'ingress' or 'egress' specifiers, I think it's perfectly clear. And every packet already carries src and dst addresses/ports. There is no confusion: src is where the packet came from and dst is where it goes to. If people want to deal with network packets they need to be aware of what src and dst means in that context.

You said:

> there is always a single interpretation for egress or ingress and cidr block combination i.e either way you look from, ingress is towards it and egress away from it.

But there's already a confusion. What do yo mean by "it"? The CIDR block? Because in our policy definition 'ingress' is away from the CIDR block and towards the "Association". That's the opposite. There is ample possibility of confusion even here.

For ingress, the traffic is TO the association. For egress it's FROM the association.

The "peer" takes on 'source' or 'destination' meaning depending on whether the policy is ingress (source) or egress (dest).